client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-pattern-pages.ts

```diff
@@ -82,18 +83,34 @@ const usePatternPages = (
 		pageSlugs = INITIAL_PAGES;
 	}
 
-	pages = pageSlugs.map( ( slug ) => pagesMapByCategory[ slug ]?.[ 0 ] ).filter( Boolean );
+	pages = pageSlugs
+		.map( ( slug ) => {
+			const firstPage = pagesMapByCategory[ slug ]?.[ 0 ];
+			if ( firstPage ) {
+				return {
+					...firstPage,
+					title: getPagePatternTitle( firstPage ),
+				};
+			}
+		} )
+		.filter( Boolean ) as Pattern[];
+
+	pagesToShow = pageCategoriesInOrder
+		.map( ( category: Category ) => {
+			const { name } = category;
+			const hasPages = name && pagesMapByCategory[ name ]?.length;
+			if ( hasPages ) {
+				const firstPage = pagesMapByCategory[ name ][ 0 ];
+				return {
+					name,
+					hasPages,
```

Comment:
[nit] I think we can now remove `hasPages` everywhere, as it is unused. And then simplify it so that it is similar to the logic in `pages` above (`const firstPage = pagesMapByCategory[ slug ]?.[ 0 ];`)

Reply:
Thanks